### DISCUSSION TITLE
Setting CARGO_TARGET_DIR

### DISCUSSION THREAD

**Discussion by StephenWakely** [helix-editor/helix#4246] [Q&A] [answered]
Is there an easy way to set the `CARGO_TARGET_DIR` env var when running rust-analyzer? 

I'd like to set it to something like `/tmp/rust-analyzer` so it doesn't lock the project directory for when I run  `cargo build`..

**Comment by the-mikedavis** [comment, marked as answer]
I think this would be enabled by https://github.com/helix-editor/helix/pull/4004

**Reply by StephenWakely** [reply]
Ah yes indeed it is. I've tested the branch from that PR and it does exactly what I need.

Many thanks.

**Comment by seanbone** [comment]
Sorry to "necro" this thread, but how does this work in practice? I have the following in my `~/.config/helix/languages.toml`:

```
[language-server.rust-analyzer.environment]
CARGO_TARGET_DIR = "target-hx"
```

However it's still building in `target`.

**Reply by kirawi** [reply]
Are you using master or 23.05? That config only works on master.

**Reply by seanbone** [reply]
Ah, I didn't realize that. I'm on 23.03 from the PPA. I'll give building from source a go, thanks.

**Reply by kirawi** [reply]
23.03 would work. What I meant was that the config syntax you're using is only for master.
